File: src/main/java/com/binarysushi/studio/lang/isml/codeInsight/tags/ISMLXmlAttributeDescriptor.java
package com.binarysushi.studio.lang.isml.codeInsight.tags;

import com.intellij.xml.impl.schema.AnyXmlAttributeDescriptor;

import java.util.HashMap;
import java.util.Map;

public class ISMLXmlAttributeDescriptor extends AnyXmlAttributeDescriptor {
    public final String tagName;
    public final Map<String, Map<String, String[]>> attrMap = new HashMap<>();

    public ISMLXmlAttributeDescriptor(String tagName, String attibuteName) {
        super(attibuteName);
        this.tagName = tagName;

        Map<String, String[]> iscacheMap = new HashMap<>();
        iscacheMap.put("status", new String[]{"on", "off"});
        iscacheMap.put("type", new String[]{"relative", "daily"});
        iscacheMap.put("hour", new String[]{"1", "12", "24"});
        iscacheMap.put("minute", new String[]{"5", "10", "15", "30", "45", "60"});
        iscacheMap.put("varyby", new String[]{"price_promotion"});
        attrMap.put("iscache", iscacheMap);

        Map<String, String[]> iscontentMap = new HashMap<>();
        iscontentMap.put("type", new String[]{"text/html", "application/xml", "application/json"});
        iscontentMap.put("encoding", new String[]{"on", "off", "html", "xml", "wml"});
        iscontentMap.put("compact", new String[]{"true", "false"});
        attrMap.put("iscontent", iscontentMap);

        Map<String, String[]> iscookieMap = new HashMap<>();
        iscookieMap.put("secure", new String[]{"on", "off"});
        attrMap.put("iscookie", iscookieMap);

        Map<String, String[]> isprintMap = new HashMap<>();
        isprintMap.put("timezone", new String[]{"SITE", "INSTANCE", "utc"});
        isprintMap.put("encoding", new String[]{"on", "off"});
        attrMap.put("isprint", isprintMap);

        Map<String, String[]> isredirectMap = new HashMap<>();
        isredirectMap.put("permanent", new String[]{"true", "false"});
        attrMap.put("isredirect", isredirectMap);

        Map<String, String[]> isremoveMap = new HashMap<>();
        isremoveMap.put("scope", new String[]{"session", "pdict", "request", "page"});
        attrMap.put("isremove", isremoveMap);

        Map<String, String[]> isselectMap = new HashMap<>();
        isselectMap.put("condition", new String[]{"true", "false"});
        isselectMap.put("encoding", new String[]{"on", "off"});
        attrMap.put("isselect", isselectMap);

        Map<String, String[]> issetMap = new HashMap<>();
        issetMap.put("scope", new String[]{"session", "request", "page"});
        attrMap.put("isset", issetMap);

        Map<String, String[]> isslot = new HashMap<>();
        isslot.put("context", new String[]{"global", "category"});
        attrMap.put("isslot", isslot);
    }

    @Override
    public String[] getEnumeratedValues() {
        String[] values = new String[0];
        Map<String, String[]> valueMap = attrMap.get(tagName);

        if (valueMap != null && valueMap.get(getName()) != null) {
            values = valueMap.get(getName());
        }

        return values;
    }
}

File: src/main/kotlin/com/binarysushi/studio/language/isml/codeInsight/tags/ISMLXmlAttributeDescriptor.kt
package com.binarysushi.studio.language.isml.codeInsight.tags

import com.intellij.xml.impl.schema.AnyXmlAttributeDescriptor

class ISMLXmlAttributeDescriptor(private val tagName: String, attributeName: String?) :
    AnyXmlAttributeDescriptor(attributeName) {

    private val attrMap: MutableMap<String, Map<String, Array<String?>>> = HashMap()
    override fun getEnumeratedValues(): Array<String?> {
        var values = arrayOfNulls<String>(0)
        val valueMap = attrMap[tagName]
        if (valueMap != null && valueMap[name] != null) {
            values = valueMap[name]!!
        }
        return values
    }

    init {
        val iscacheMap: MutableMap<String, Array<String?>> = HashMap()
        iscacheMap["status"] = arrayOf("on", "off")
        iscacheMap["type"] = arrayOf("relative", "daily")
        iscacheMap["hour"] = arrayOf("1", "12", "24")
        iscacheMap["minute"] = arrayOf("5", "10", "15", "30", "45", "60")
        iscacheMap["varyby"] = arrayOf("price_promotion")
        attrMap["iscache"] = iscacheMap

        val iscontentMap: MutableMap<String, Array<String?>> = HashMap()
        iscontentMap["type"] = arrayOf("text/html", "application/xml", "application/json")
        iscontentMap["encoding"] = arrayOf("on", "off", "html", "xml", "wml")
        iscontentMap["compact"] = arrayOf("true", "false")
        attrMap["iscontent"] = iscontentMap

        val iscookieMap: MutableMap<String, Array<String?>> = HashMap()
        iscookieMap["secure"] = arrayOf("on", "off")
        attrMap["iscookie"] = iscookieMap

        val isprintMap: MutableMap<String, Array<String?>> = HashMap()
        isprintMap["timezone"] = arrayOf("SITE", "INSTANCE", "utc")
        isprintMap["encoding"] = arrayOf("on", "off")
        attrMap["isprint"] = isprintMap

        val isredirectMap: MutableMap<String, Array<String?>> = HashMap()
        isredirectMap["permanent"] = arrayOf("true", "false")
        attrMap["isredirect"] = isredirectMap

        val isremoveMap: MutableMap<String, Array<String?>> = HashMap()
        isremoveMap["scope"] = arrayOf("session", "pdict", "request", "page")
        attrMap["isremove"] = isremoveMap

        val isselectMap: MutableMap<String, Array<String?>> = HashMap()
        isselectMap["condition"] = arrayOf("true", "false")
        isselectMap["encoding"] = arrayOf("on", "off")
        attrMap["isselect"] = isselectMap

        val issetMap: MutableMap<String, Array<String?>> = HashMap()
        issetMap["scope"] = arrayOf("session", "request", "page")
        attrMap["isset"] = issetMap

        val isslot: MutableMap<String, Array<String?>> = HashMap()
        isslot["context"] = arrayOf("global", "category")
        attrMap["isslot"] = isslot
    }
}